torch/csrc/api/include/torch/serialization.h

```diff
@@ -217,25 +217,24 @@ void load(Archive& archive, torch::Tensor& tensor) {
   archive(CEREAL_NVP(backendId), CEREAL_NVP(sizes));
 
   at::Backend backend = ::torch::detail::backendFromId(backendId);
-  if (!tensor.defined() || tensor.data().type().scalarType() != type) {
+  if (!tensor.defined() || tensor.dtype() != type) {
     tensor = torch::empty({}, torch::getType(backend, type));
   }
   tensor.data().resize_(sizes);
 
   if (tensor.type().is_cuda()) {
     // should actually use cudamemcpy probably
-    auto cputensor = torch::empty(sizes, tensor.data().type().scalarType());
+    auto cputensor = torch::empty(sizes, tensor.dtype());
     agimpl::loadBinary(
         archive,
         cputensor.data_ptr(),
         cputensor.numel() * cputensor.type().elementSizeInBytes());
-    tensor.data().copy_(cputensor.data());
+    tensor.copy_(cputensor);
```

Comment:
I think the error I encountered was that this would mark the original tensor "already used in an in-place operation", and then future forwards on any loaded model would not work. Unless I'm misunderstanding what is going on here...